pkg/vm/ref_counter.go

```diff
@@ -5,27 +5,27 @@ import (
 )
 
 // refCounter represents reference counter for the VM.
-type refCounter struct {
-	items map[stackitem.Item]int
-	size  int
+type refCounter int
+
+type rc interface {
+	IncRC() int
+	DecRC() int
 }
 
 func newRefCounter() *refCounter {
-	return &refCounter{
-		items: make(map[stackitem.Item]int),
-	}
+	return new(refCounter)
 }
 
 // Add adds an item to the reference counter.
 func (r *refCounter) Add(item stackitem.Item) {
 	if r == nil {
 		return
 	}
-	r.size++
+	*r++
 
 	switch item.(type) {
 	case *stackitem.Array, *stackitem.Struct, *stackitem.Map:
-		if r.items[item]++; r.items[item] > 1 {
+		if item.(rc).IncRC() > 1 {
```

Comment:
What's going to be faster, current code or
```
rcItem, ok := item.(rc)
if !ok || rcItem.IncRC() > 1 {
    return
}
switch t := item.(type) {
...
```
?

Reply:
It is faster by 3% (and by another 3% if we split the interface). Fixed, also looks simpler.